linux_os/guide/system/permissions/files/permissions_within_important_dirs/root_permissions_syslibrary_files/oval/shared.xml

```diff
@@ -0,0 +1,28 @@
+<def-group>
+  <definition class="compliance" id="root_permissions_syslibrary_files" version="2">
+    {{{ oval_metadata("
+        Checks that system-wide library files in /lib, /lib64, /usr/lib, /usr/lib64
+        are owned by root.
+      ") }}}
+    <criteria >
+      <criterion test_ref="test_root_permissions_for_syslibrary_files" />
+    </criteria>
+  </definition>
+
+  <unix:file_test  check="all" check_existence="none_exist" comment="test if system-wide files have root permissions" id="test_root_permissions_for_syslibrary_files" version="1">
+    <unix:object object_ref="root_permissions_for_system_wide_library_files" />
+  </unix:file_test>
+
+  <unix:file_object comment="system-wide directories" id="root_permissions_for_system_wide_library_files" version="1">
+    <!-- Checks that system-wide library files in /lib, /lib64, /usr/lib, /usr/lib64
+        are owned by root. -->
+    <unix:behaviors recurse_direction="down" />
+    <unix:path operation="pattern match">^\/lib(64)?|^\/usr\/lib(64)?</unix:path >    
```

Comment:
Unfortunately it's not possible to use recursion with pattern match:

`
oval:ssg-root_permissions_for_system_wide_library_files:obj:1 - the recurse_direction behavior MUST not be used when a pattern match is used with a path entity.
`

I believe you have to create an object for each of the directories you are looking for (`/lib`, `/lib64`, `/usr/lib` and `/usr/lib64`)

Reply:
also we might wanna double check if recursion is also needed in other similar rules to this one.

Reply:
> also we might wanna double check if recursion is also needed in other similar rules to this one. Maybe other rules will have the same behavior, such as the one being proposed here: https://github.com/ComplianceAsCode/content/pull/7002